src/Adapter/Attachment/CommandHandler/BulkDeleteAttachmentsHandler.php

```diff
@@ -0,0 +1,54 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Attachment\CommandHandler;
+
+use PrestaShop\PrestaShop\Adapter\Attachment\AbstractAttachmentHandler;
+use PrestaShop\PrestaShop\Core\Domain\Attachment\Command\BulkDeleteAttachmentsCommand;
+use PrestaShop\PrestaShop\Core\Domain\Attachment\CommandHandler\BulkDeleteAttachmentHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Attachment\Exception\DeleteAttachmentException;
+
+/**
+ * Class BulkDeleteAttachmentsHandler
+ */
+final class BulkDeleteAttachmentsHandler extends AbstractAttachmentHandler implements BulkDeleteAttachmentHandlerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(BulkDeleteAttachmentsCommand $command)
+    {
+        foreach ($command->getAttachmentIds() as $attachmentId) {
+            $attachment = $this->getAttachment($attachmentId);
+
+            if (!$this->deleteAttachment($attachment)) {
+                throw new DeleteAttachmentException(
```

Comment:
Why if only the last one failed?
The response will return an error while everything instead of one id weren't deleted? :thinking:

Reply:
I see there is an unresolved discussion about this topic https://github.com/PrestaShop/PrestaShop/issues/14179

Reply:
What was decided for this behavior?